### PR TITLE
SSCS-11400 - Set Hearing Options For Interpreter Language When Not Selected

### PR DESCRIPTION
**Pull request by DGordonV1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-11400

### Change description ###

SSCS-11400 - Set Hearing Options For Interpreter Language When Not Selected

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by AndreiTrushkov — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java [hunk 1]
```diff
@@ -80,17 +81,20 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
             hearingMessageHelper.sendListAssistCreateHearingMessage(sscsCaseData.getCcdCaseId());
         }
 
+        HearingOptions hearingOptions = HearingOptions.builder().build();
+        if (sscsCaseData.getAppeal().getHearingOptions() != null) {
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java:85]:
```suggestion
        HearingOptions caseDataHearingOptions = sscsCaseData.getAppeal().getHearingOptions();
hearingOptions = caseDataHearingOptions != null ? caseDataHearingOptions : HearingOptions.builder().build()
```

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

**Review by jarekPierchala — DISMISSED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java [hunk 2]
```diff
@@ -80,17 +81,18 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
...
+        HearingOptions hearingOptions = caseDataHearingOptions != null ? caseDataHearingOptions : HearingOptions.builder().build();
+
         if (adjournment.getInterpreterRequired() != null) {
-            HearingOptions hearingOptions = HearingOptions.builder().build();
-            if (sscsCaseData.getAppeal().getHearingOptions() != null) {
-                hearingOptions = sscsCaseData.getAppeal().getHearingOptions();
-            }
             DynamicList interpreterLanguage = adjournment.getInterpreterLanguage();
-            hearingOptions.setLanguages(nonNull(interpreterLanguage.getValue()) ? interpreterLanguage.getValue().getLabel() : "");
+            hearingOptions.setLanguages(nonNull(interpreterLanguage.getValue()) ? interpreterLanguage.getValue().getLabel() : NO.getValue());
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java:89]: Need a test for this
image

**Review by amoslv1 — CHANGES_REQUESTED** [review]
Add a test case for "No" result

**Review by AndreiTrushkov — DISMISSED** [review]

**Review by amoslv1 — DISMISSED** [review]

**Comment by nickhill111** [comment]
Code moved to SSCS-10865

**Pull request closed without merging**